src/tpm2.c

```diff
@@ -5072,34 +5074,39 @@ UINT16 TPM2_GetVendorID(void)
 #define LINE_LEN 16
 void TPM2_PrintBin(const byte* buffer, word32 length)
 {
-    word32 i;
+    word32 i, sz;
     char line[80];
 
     if (!buffer) {
         printf("\tNULL");
         return;
     }
 
-    sprintf(line, "\t");
+    while (length > 0) {
+        sz = length;
+        if (sz > LINE_LEN)
+            sz = LINE_LEN;
+        sprintf(line, "\t");
```

Comment:
By using a temp `char*` pointer, I think `sprintf()` is a little less error prone in case someone changes the format strings. Each `sprintf` then `+=` to the pointer for fewer equations.

```suggestion
        line_ptr = line;
        line_ptr += sprintf(line, "\t");
```

Reply:
Excellent suggestion. Fixes pushed.